experimental/cuda2/cuda_driver.c

```diff
@@ -0,0 +1,448 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <stdint.h>
+#include <string.h>
+
+#include "experimental/cuda2/api.h"
+#include "experimental/cuda2/dynamic_symbols.h"
+#include "experimental/cuda2/status_util.h"
+#include "iree/base/api.h"
+#include "iree/base/assert.h"
+#include "iree/base/tracing.h"
+#include "iree/hal/api.h"
+
+// Maximum device name length supported by the CUDA HAL driver.
+#define IREE_HAL_CUDA_MAX_DEVICE_NAME_LENGTH 128
+
+// Utility macros to convert between CUDevice and iree_hal_device_id_t.
+#define CUDEVICE_TO_DEVICE_ID(device) (iree_hal_device_id_t)((device) + 1)
+#define DEVICE_ID_TO_CUDEVICE(device_id) (CUdevice)((device_id)-1)
```

Comment:
weird that clang-format removes the spaces on the -1?

Reply:
Yes for this case.